api/src/opentrons/motion_planning/deck_conflict.py

```diff
@@ -274,20 +283,31 @@ def _create_ot2_restrictions(
 
 
 def _create_flex_restrictions(
-    item: DeckItem, location: DeckSlotName
+    item: DeckItem, location: Union[DeckSlotName, StagingSlotName]
 ) -> List[_DeckRestriction]:
-    restrictions: List[_DeckRestriction] = []
-
-    if location not in _FIXED_TRASH_SLOT:
-        restrictions.append(
-            _NothingAllowed(
-                location=location,
-                source_item=item,
-                source_location=location,
-            )
+    restrictions: List[_DeckRestriction] = [
+        _NothingAllowed(
+            location=location,
+            source_item=item,
+            source_location=location,
         )
+    ]
+
+    # Magnetic block module does not have staging slot restriction
+    if isinstance(item, (HeaterShakerModule, OtherModule)):
+        if isinstance(location, StagingSlotName):
+            raise DeckConflictError("Cannot have a module loaded on a staging slot.")
```

Comment:
Nitpick to match what I see in the manual.

```suggestion
            raise DeckConflictError("Cannot have a module loaded on a staging area slot.")
```